### PR TITLE
Rework downstream E2E tests

### PR DESCRIPTION
**Pull request by chmouel**
Separate the test between privileged and not
Do some magic for the privileged test by adding a serviceAccount which has the
privileged scc.

### CODE REVIEW

File: openshift/e2e-tests-openshift.sh [hunk 1]
```diff
@@ -1,6 +1,63 @@
...
+# in_array function: https://www.php.net/manual/en/function.in-array.php :-D
+function in_array() {
+    param=$1;shift
+    for elem in $@;do
+        [[ "$param" == "$elem" ]] && return 0;
+    done
+    return 1
+}
+
+# Run the non privileged tests
```
Comment (piyush-garg) [inline openshift/e2e-tests-openshift.sh]: Run the privileged tests - confusing

**Comment by chmouel** [comment]
Bad copy and paste :D

File: openshift/Makefile [hunk 2]
```diff
@@ -16,5 +16,5 @@ test-e2e-downstream-stable:
 
 test-e2e-downstream:
 	@env RELEASE_YAML=$(RELEASE_YAML) ./e2e-tests-openshift.sh
-	@env LOCAL_CI_RUN=true TEST_YAML_IGNORES="kaniko s2i buildah jib-maven" ../test/e2e-tests.sh
```
Comment (piyush-garg) [inline openshift/Makefile:19]: I think we need to skip these in downstream

Comment (chmouel) [reply openshift/Makefile:19]: the whole point of this patch is to make them working downstream.... https://github.com/openshift/tektoncd-catalog/pull/57#issuecomment-555528899

**Comment by chmouel** [comment]
worked nicely https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_tektoncd-catalog/60/pull-ci-openshift-tektoncd-catalog-release-next-e2e/84/artifacts/e2e/container-logs/test.log in #60

**Review by piyush-garg — APPROVED** [review]
/lgtm

Nice work :bowing_man:

**Pull request merged**